openedx/core/lib/xblock_keys/bundle_def.py

```diff
@@ -0,0 +1,139 @@
+import base64
+from uuid import UUID
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import DefinitionKey
+
+
+class BundleDefinitionLocator(DefinitionKey):
+    """
+    Implementation of the DefinitionKey type, for XBlock content stored in
+    Blockstore bundles (.olx files).
+
+    Each BundleDefinitionLocator holds the following data
+        bundle UUID (e.g. fc4d4d8d-60b3-4aad-b9f1-4ea738505d13)
+        OLX file path (e.g. /unit-welcome/welcome.olx)
+        Block type (of some block within the OLX file, e.g. 'html')
+        Block definition ID (of some block within the OLX file, e.g. 'intro')
```

Comment:
A couple of questions:
1. If an OLX file A references another OLX file B, which path is used for the locator of the root block of OLX file B?
2. Can an OLX file C reference a nested block in OLX file D? If yes, how would the reference be serialized in the OLX?

Reply:
1. I haven't yet figured out how OLX files will reference other OLX files, but I wasn't planning on proposing using these Definition Keys for that (see comment below). Instead I was thinking we would use a very simple scheme as shown in [bundles.rst](https://github.com/open-craft/blockstore/blob/master/docs/bundles.rst) such as `bundle://videos_lectures/vid/lecture1.olx`

> Can an OLX file C reference a nested block in OLX file D?

2. That's an open question. It would be nice if we can say no, but I'm guessing we're going to have to find a way to support that use case.